opensrp-pnc/src/main/java/org/smartregister/pnc/utils/PncUtils.java

```diff
@@ -376,4 +378,201 @@ public static void processEvents(@NonNull List<PncEventClient> pncEventClients)
+            HashMap<String, String> data = pncBaseDetails.getProperties();
+
+            if ("1".equals(data.get(PncConstants.JsonFormKeyConstants.OUTCOME_SUBMITTED))) {
+
+                String deliveryDateStr = data.get(PncConstants.FormGlobalConstants.DELIVERY_DATE);
+                LocalDate deliveryDate = LocalDate.parse(deliveryDateStr, DateTimeFormat.forPattern("dd-MM-yyyy"));
+                VisitScheduler pncVisitScheduler = new PncVisitScheduler(deliveryDate, baseEntityId);
+
+                if (pncVisitScheduler.getStatus() == VisitStatus.PNC_DUE) {
+                    button.setText(R.string.pnc_due);
+                    button.setTag(R.id.BUTTON_TYPE, R.string.pnc_due);
+                    button.setTextColor(ContextCompat.getColor(button.getContext(), R.color.due_color));
+                    button.setBackground(ContextCompat.getDrawable(button.getContext(), R.drawable.pnc_btn_due_bg));
+                } else if (pncVisitScheduler.getStatus() == VisitStatus.PNC_OVERDUE) {
+                    button.setText(R.string.pnc_due);
+                    button.setTag(R.id.BUTTON_TYPE, R.string.pnc_overdue);
+                    button.setTextColor(ContextCompat.getColor(button.getContext(), R.color.white));
+                    button.setBackgroundColor(ContextCompat.getColor(button.getContext(), R.color.overdue_color));
+                } else if (pncVisitScheduler.getStatus() == VisitStatus.RECORD_PNC) {
+                    button.setText(R.string.record_pnc);
+                    button.setTag(R.id.BUTTON_TYPE, R.string.record_pnc);
+                    button.setTextColor(ContextCompat.getColor(button.getContext(), R.color.due_color));
+                    button.setBackground(ContextCompat.getDrawable(button.getContext(), R.drawable.pnc_btn_due_bg));
+
+                } else if (pncVisitScheduler.getStatus() == VisitStatus.PNC_DONE_TODAY) {
+                    button.setText(R.string.pnc_done_today);
+                    button.setTag(R.id.BUTTON_TYPE, R.string.pnc_done_today);
+                    button.setTextColor(ContextCompat.getColor(button.getContext(), R.color.dark_grey));
+                    button.setBackground(ContextCompat.getDrawable(button.getContext(), R.drawable.pnc_btn_done_today));
+                } else if (pncVisitScheduler.getStatus() == VisitStatus.PNC_CLOSE) {
+                    button.setText(R.string.pnc_close);
+                    button.setTag(R.id.BUTTON_TYPE, R.string.pnc_close);
+                }
+            }
+        }
+    }
+
+    public static void addGlobals(String baseEntityId, JSONObject form) {
+
+        String q1 = "SELECT * FROM ec_client WHERE base_entity_id = '" + baseEntityId + "'";
+        String q2 = "SELECT * FROM pnc_registration_details WHERE base_entity_id = '" + baseEntityId + "'";
+        String q3 = "SELECT dob AS baby_dob FROM pnc_baby WHERE mother_base_entity_id = '" + baseEntityId + "'";
```

Comment:
use a join for these queries and specify the columns you need.

Reply:
done